recipes/pupnp/all/conanfile.py

```diff
@@ -41,95 +49,74 @@ class PupnpConan(ConanFile):
             raise ConanInvalidConfiguration("Visual Studio not supported yet in this recipe")
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
-
     def build_requirements(self):
-        self.build_requires("libtool/2.4.6")
-        self.build_requires("pkgconf/1.7.4")
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        self.tool_requires("libtool/2.4.7")
+        if not self.conf.get("tools.gnu:pkg_config", default=False, check_type=str):
+            self.tool_requires("pkgconf/1.9.5")
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_autotools(self):
-        if not self._autotools:
-            args = [
-                "--enable-static=%s" % ("no" if self.options.shared else "yes"),
-                "--enable-shared=%s" % ("yes" if self.options.shared else "no"),
-                "--disable-samples",
-            ]
-
-            def enable_disable(opt):
-                what = "enable" if getattr(self.options, opt) else "disable"
-                return "--{}-{}".format(what, opt)
-
-            args.extend(
-                map(
-                    enable_disable,
-                    (
-                        "ipv6",
-                        "reuseaddr",
-                        "webserver",
-                        "client",
-                        "device",
-                        "largefile",
-                        "tools",
-                        "debug"
-                    ),
-                )
-            )
-
-            args.append("--%s-blocking_tcp_connections" % ("enable" if getattr(self.options, "blocking-tcp") else "disable"))
-
-            self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-            self._autotools.configure(configure_dir=self._source_subfolder, args=args)
-        return self._autotools
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        tc = AutotoolsToolchain(self)
+        features = {}
+        features["samples"] = False
+        features["blocking_tcp_connections"] = self.options["blocking-tcp"]
+        for opt in ("ipv6", "reuseaddr", "webserver", "client", "device", "largefile", "tools", "debug"):
+            features[opt] = self.options[opt]
+        for feature, enabled in features.items():
+            what = "enable" if enabled else "disable"
+            tc.configure_args.append(f"--{what}-{feature}")
```

Comment:
Very elegant, nice!

Reply:
@valgur somethings broken here, because it call all switches with --disable (see logs)
Found it here: https://github.com/gerbera/gerbera/actions/runs/5919696297/job/16050215927#step:7:1

Reply:
Oof, thank you for reporting this! Created a PR: #19322